README.md

```diff
@@ -27,31 +29,123 @@ npm install @dojo/widgets
 
-Dojo 2 widgets are fully themeable. In a future release we will ship with several official Dojo 2 themes that can be applied to the widgets (all themes will live in a separate repo).
+[TabPane](src/tabpane/README.md)
 
-Dojo 2 widgets are fully internationalisable (`i18n`)
+### Misc widgets
+[Dialog](src/dialog/README.md)
 
 ## Conventions
 
+### EventHandlers
+
+You can register event handlers that get called when the corresponding events occur, by passing the handlers into a widget, in their `properties`.
+The naming convention for event handlers is as follows:
+
+- if the parent of the widget has the power to decide *if* an event is successful, i.e. can cancel the event, then the child widget will call an event handler in the following format:
+
+`onRequest[X]`, e.g. for a `close` event, the event handler called by the child widget must be called `onRequestClose`
+
+Here the child widget is requesting that the `close` event take place.
+
+- for events that will occur regardless of child/parent interaction, then the `Request` naming convention is dropped:
+
+`on[X]`, e.g. for a `dismiss` event, then event handler called by the child widget must be called `onDismiss`
+
+### Controlled vs. uncontrolled form widgets
+Currently, all of our *form* widgets are controlled widgets.
+This means that a parent widget is responsible to the state of the child widget.
+
+e.g.
+
+```ts
+[in WidgetFoo.ts]
+
+w(CheckBox, {}, []);
+```
+Here, `WidgetFoo` is the parent widget, which in turn, creates a child widget of type `CheckBox`.
+In this instance, the `CheckBox` is controlled by `WidgetFoo`, therefore the `CheckBox` is known as a 'controlled' widget.
+Controlled widgets are also known as 'stateless' widgets, because they do not hold internal state; their state is passed in as part of their `properties`.
+
 ### Icons
 
 We use [font awesome](http://fontawesome.io/) for icons.
 Where a theme requires specific icons that are not part of the Font Awesome set, then those themes will ship their own icons.
 
 ### Coding conventions
 
-`px vs em` - we specify font sizes in `px`. When creating a widget, spacing (margin, padding) should be specified using `px` unless the design calls for proportional spacing, in which case `em` can be used.
+`px vs. em` - we specify font sizes in `px`.
+When creating a widget, spacing (margin, padding) should be specified using `px` unless the design calls for proportional spacing, in which case `em` can be used.
+
+
+## How to customize a widget
+
+There are may ways in which you can customize the behaviour and appearance of Dojo 2 widgets.
+See the [`widget-core`](https://github.com/dojo/widget-core) README for examples of how to customize the theme or a specific CSS class of a widget.
 
+Or can you write your own widget that extends an official widget.
+
+### Extending widgets
+
+If you want to extend one of the widgets provide with Dojo 2, then since all our widgets are Classes, you can simply extend it:
```

Comment:
"If you want to extend one of the widgets provide with Dojo 2, then since all our widgets are Classes, you can simply extend it" perhaps something like -> Because all Dojo 2 widgets are Classes, you can simply extend the Class to add or change its behavior.